src/Pages/Speaker/Speaker.js

```diff
@@ -34,24 +38,57 @@ function SpeakersPage(props) {
     }
   };
 
+  const updateDisplay = (queuedSongs) => {
+    if (queuedSongs.length > 0) {
+      setNextUpSong(queuedSongs[0]);
+      setQueue(queuedSongs.slice(1));
+    } else {
+      setNextUpSong(null);
+      setQueue([]);
+    }
+  };
+
   const getQueuedSongs = async () => {
     const songList = await queued(props.user.token);
     if (songList.error) {
       setError('Unable to reach speaker: ' + String(songList.responseData));
     }
     if (Array.isArray(songList.responseData)) {
-      setQueuedSongs(songList.responseData);
+      let q = songList.responseData;
+      setQueuedSongs(q);
+      updateDisplay(q);
     }
   };
 
-
   const modifySpeakerWrapper = async (modifier) => {
     const result = await modifier(props.user.token);
     if (result.error) {
       setError(String(result.responseData));
     }
   };
 
+  const togglePlayback = async () => {
+    if (isPlaying) {
+      await modifySpeakerWrapper(pause);
+    } else {
+      await modifySpeakerWrapper(resume);
+    }
+    setIsPlaying(!isPlaying);
+  };
+
+  const handleVolumeChange = async (volume) => {
+    setVolumeState(volume); // Update local volume state
+    const result = await setVolume(volume, props.user.token);
+  };
```

Comment:
will this get called any time the slider changes value, or when the user is done dragging the slider?

Reply:
When user is done dragging slider

Reply:
now it slike theat